src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,91 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String DEFAULT_LOGIN = "turtle54@gmail.com";
+    private static final String DEFAULT_PASSWORD = "abcde12345";
+    private static final int DEFAULT_AGE = 21;
+    private User user;
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
+        user.setLogin(DEFAULT_LOGIN);
+        user.setPassword(DEFAULT_PASSWORD);
+        user.setAge(DEFAULT_AGE);
+    }
+
+    @Test
+    void register_userLoginExistsInStorage_notOk() {
+        Storage.people.add(user);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_validUser_registrationSuccess() {
+        registrationService.register(user);
+        assertEquals(1, Storage.people.size());
+        assertEquals(user, Storage.people.get(0));
+    }
+
+    @Test
+    void register_LoginIsLessThanMinChars_notOk() {
+        List<String> invalidLogins = Arrays.asList(null, ""," ","abc", "abcde");
+        for (String invalidLogin: invalidLogins) {
```

Comment:
```suggestion
        for (String invalidLogin : invalidLogins) {
```
regularly do Ctrl Alt L